ceno_zkvm/src/circuit_builder.rs

```diff
@@ -41,24 +41,10 @@ impl NameSpace {
     }
 
     pub(crate) fn compute_path(&self, this: String) -> String {
-        let ns = self.get_namespaces();
-        if this.chars().any(|a| a == '/') {
+        if this.chars().contains(&'/') {
             panic!("'/' is not allowed in names");
         }
-
-        let mut name = String::new();
-
-        let mut needs_separation = false;
-        for ns in chain!(ns, once(&this)) {
-            if needs_separation {
-                name += "/";
-            }
-
-            name += ns;
-            needs_separation = true;
-        }
-
-        name
+        chain!(self.get_namespaces(), once(&this)).join("/")
```

Comment:
This is mostly just realising that `join` exists and does exactly what we implemented above manually.